modules/ipi-install-configuring-host-network-interfaces-in-the-install-config.yaml-file.adoc

```diff
@@ -2,62 +2,119 @@
-            password: <password>
+            address: redfish+http://_<out_of_band_ip>_/redfish/v1/Systems/
+            username: _<user>_
+            password: _<password>_
             disableCertificateVerification: null
-          bootMACAddress: <NIC1_mac_address>
+          bootMACAddress: _<NIC1_mac_address>_
           bootMode: UEFI
           rootDeviceHints:
             deviceName: "/dev/sda"
           networkConfig: <1>
             interfaces:
-            - name: <NIC1_name>
+            - name: _<NIC1_name>_
               type: ethernet
               state: up
               ipv4:
                 address:
-                - ip: "<IP_address>"
+                - ip: "_<IP_address>_"
                   prefix-length: 24
                 enabled: true
             dns-resolver:
               config:
                 server:
-                - <DNS_IP_address>
+                - _<DNS_IP_address>_
             routes:
               config:
               - destination: 0.0.0.0/0
-                next-hop-address: <IP_address>
-                next-hop-interface: <NIC1_name>
+                next-hop-address: _<IP_address>_
+                next-hop-interface: _<NIC1_name>_
 ----
 <1> Add NMState YAML syntax to configure host interfaces.
++
+When configuring static IP addresses, be aware that while the cluster's nodes can be assigned a static IP address as part of the `install-config` file, the bootstrap VM cannot be assigned a static IP address. In the case of deploying {product-title}, you must configure it using Ignition, as shown in the following example:
 
-[TIP]
-====
-Consider saving the `networkConfig` YAML syntax to a file and testing it using the NMState command line interface before including it in the `install-config.yaml` file, because the installer will not check the NMState YAML syntax. Execute `nmstatectl gc <yaml-config>` to test the syntax. Errors in the YAML syntax might result in a failure to apply the network configuration. Additionally, maintaining the validated YAML syntax is useful when applying changes using Kubernetes NMState after deployment or when expanding the cluster.
-====
+. Create a NetworkManager `keyFile` for the external NIC of the boostrap VM:
++
+.Example NetworkManager keyFile
+[source,text]
+----
+  BOOTSTRAP_CONFIG="[connection]
+  type=ethernet
+  interface-name=ens3
+  [ethernet]
+  [ipv4]
+  method=manual
+  addresses=${BOOTSTRAP_IP}/${BOOTSTRAP_PREFIX}
+  gateway=${PROVISIONING_HOST_EXTERNAL_IP}
+  dns=${PROVISIONING_HOST_EXTERNAL_IP}"
+----
++
+. Update the generated `bootstrap.ign` file with the following changes:
++
+.Example
+[source,terminal]
+----
+$ cat > bootstrap_network_config.ign << EOF
+      {
+        "path": "/etc/NetworkManager/system-connections/ens3.nmconnection",
+        "mode": 384,
+        "contents": {
+        "source": "data:text/plain;charset=utf-8;base64,$(echo "${BOOTSTRAP_CONFIG}" | base64 -w 0)"
+        }
+      }
+  EOF
```

Comment:
Are the lines that follow this `EOF` here supposed to be additional individual commands to be entered, or are they part of the bootstrap file?

Reply:
They are additional individual commands.

Reply:
This should be four commands, but:
1. They are divided wrong here. The third command should be starting with jq
2. there is a confusion here between the 'tmp.sh' sript, which needs to be run, and the bootstrap.ign file, which is re-created by the 'jq' command. (So, basically, all of the file names should stay the same as the ones provided by Steve, except tmp.sh, which can be changed to something more generic)

Reply:
Updated as suggested.